src/bitmap/utils/zip_validity.rs

```diff
@@ -101,10 +101,13 @@ where
     V: Iterator<Item = bool>,
 {
     /// Returns a new [`ZipValidity`]
-    pub fn new(values: I, validity: Option<V>) -> Self {
+    pub fn new(values: I, validity: Option<V>, null_count: Option<usize>) -> Self {
```

Comment:
instead of passing the option here, could we pass

```rust
array.validity().and_then(|validity| (validity.unset_bits() > 0).then(|| validity.iter()))
```

to `Zip::new` instead? This keeps `Zip::new` with a single responsibility.

Reply:
Yeap, much cleaner as well. That I didn't think of that one.